core/domain/services/contexts/RequestTypeContextDetector.php

```diff
@@ -231,8 +246,8 @@ private function isFeedRequest()
     private function uriPathMatches($component)
     {
         $request_uri = $this->request->requestUri(true);
-        $parts       = explode('?', $request_uri);
-        $path        = trim(reset($parts), '/');
+        $parts = explode('?', $request_uri);
+        $path = trim(reset($parts), '/');
```

Comment:
This also matches `https://example.com/blog/why-you-should-use-graphql-for-wp` 😄

Reply:
the code you are highlighting doesn't even look at the parameter you want to compare against, it is simply preparing the URL for comparison which happens on the last line

here's the entire method:

```php
    private function uriPathMatches($component)
    {
        $request_uri = $this->request->requestUri(true);
        $parts       = explode('?', $request_uri);
        $path        = trim(reset($parts), '/');
        return strpos($path, $component) === 0;
    }
```

- the `$component` parameter is the URL chunk you want to match, like `graphql`

- `$request_uri` is the request portion of the URI relative to the WP root.
  from the phpdocs of `Request::requestUri()`:
  > If `home_url()` is "http://mysite.com/wp/", and a request comes to "http://mysite.com/wp/wp-json", setting `$relativeToWpRoot=true` will return "/wp-json", whereas `$relativeToWpRoot=false` will return "/wp/wp-json/"
  For a URL like `example.com/blog/why-you-should-use-graphql-for-wp?some-param=123`
  `$request_uri` would be `/blog/why-you-should-use-graphql-for-wp?some-param=false`

- `$parts` simply splits the URI at the point where the parameters begin and dumps the two parts into an array

- `reset($parts)` grabs the first element from the above array and `trim( xxxxx, '/')` removes any backslashes. So in our example `$path` would now be `blog/why-you-should-use-graphql-for-wp`. At this point, no attempt to match this path with a string has occurred, that happens in the next step

- `strpos($path, $component) === 0` checks if the position of the `$component` string within the `$path` string is `0`, ie: it **only** returns `true` if the `$path` **begins** with `$component`. `$component` can be anywhere else in the string but if it's not at the very beginning, then it returns `false`

- passing `graphql` to this method would then be checking if the current request URI starts with `graphql`, and since `blog/why-you-should-use-graphql-for-wp` does not (`graphql` is at position 24), it would return `false`

- if the current URL was `example.com/graphql` then it would return `true`, OR if `$component` was "blog" and the URL was `example.com/blog/why-you-should-use-graphql-for-wp` then it would also return `true`